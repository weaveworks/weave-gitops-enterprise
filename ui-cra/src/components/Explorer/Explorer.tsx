import { Flex, Icon, IconType } from '@weaveworks/weave-gitops';
// @ts-ignore
import { IconButton } from '@material-ui/core';
import { Alert } from '@material-ui/lab';
import _ from 'lodash';
import { useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import styled from 'styled-components';
import { Facet } from '../../api/query/query.pb';
import { useListFacets, useQueryService } from '../../hooks/query';
import ExplorerTable from './ExplorerTable';
import FilterDrawer from './FilterDrawer';
import Filters from './Filters';
import PaginationControls from './PaginationControls';
import QueryInput from './QueryInput';
import QueryStateChips from './QueryStateChips';
import { QueryStateManager, URLQueryStateManager } from './QueryStateManager';
import { QueryStateProvider, columnHeaderHandler } from './hooks';

type Props = {
  className?: string;
  category?: 'automation' | 'source';
  enableBatchSync?: boolean;
  manager?: QueryStateManager;
};

function Explorer({ className, category, enableBatchSync, manager }: Props) {
  const navigate = useNavigate();
  const location = useLocation();
  if (!manager) {
    manager = new URLQueryStateManager(navigate, location);
  }

  const [filterDrawerOpen, setFilterDrawerOpen] = useState(false);
  const { data: facetsRes } = useListFacets();
  const queryState = manager.read();
  const setQueryState = manager.write;

  const { data, error } = useQueryService({
    terms: queryState.terms,
    filters: queryState.filters,
    limit: queryState.limit,
    offset: queryState.offset,
    orderBy: queryState.orderBy,
    ascending: queryState.orderAscending,
    category,
  });

  const filteredFacets = filterFacetsForCategory(facetsRes?.facets, category);

  return (
    <QueryStateProvider manager={manager}>
      <div className={className}>
        {error && <Alert severity="error">{error.message}</Alert>}
        <Flex align wide end>
          <QueryStateChips />
          <IconButton onClick={() => setFilterDrawerOpen(!filterDrawerOpen)}>
            <Icon size="normal" type={IconType.FilterIcon} />
          </IconButton>
        </Flex>
        <Flex wide>
          <ExplorerTable
            queryState={queryState}
            rows={data?.objects || []}
            onColumnHeaderClick={columnHeaderHandler(queryState, setQueryState)}
            enableBatchSync={enableBatchSync}
          />

          <FilterDrawer
            onClose={() => setFilterDrawerOpen(false)}
            open={filterDrawerOpen}
          >
            <QueryInput />

            <Filters facets={filteredFacets || []} />
          </FilterDrawer>
        </Flex>

        <PaginationControls
          queryState={queryState}
          setQueryState={setQueryState}
          count={data?.objects?.length || 0}
        />
      </div>
    </QueryStateProvider>
  );
}

export default styled(Explorer).attrs({ className: Explorer.name })`
  width: 100%;
`;

const categoryKinds = {
  automation: ['Kustomization', 'HelmRelease'],
  source: [
    'GitRepository',
    'HelmRepository',
    'Bucket',
    'HelmChart',
    'OCIRepository',
  ],
};

function filterFacetsForCategory(
  facets?: Facet[],
  category?: 'automation' | 'source',
): Facet[] {
  if (!category) {
    return _.sortBy(facets, 'field') as Facet[];
  }

  const withoutKind = _.filter(facets, facet => {
    return facet.field?.toLowerCase() !== 'kind';
  });

  const kindFacets = _.map(categoryKinds[category], k => k.toLowerCase());

  withoutKind.unshift({
    field: 'Kind',
    values: kindFacets,
  });

  return _.sortBy(withoutKind, 'field');
}
