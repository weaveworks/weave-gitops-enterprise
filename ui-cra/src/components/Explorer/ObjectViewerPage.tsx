import { Page } from '@weaveworks/weave-gitops';
import styled from 'styled-components';
import GenericObjectViewer, { ObjectViewerProps } from './GenericObjectViewer';

function ObjectViewerPage({ className, ...props }: ObjectViewerProps) {
  return (
    <Page
      path={[
        { label: 'Explorer', url: '/explorer' },
        {
          label: `${props.kind}/${props.name}`,
        },
      ]}
    >
      <GenericObjectViewer {...props} />
    </Page>
  );
}

export default styled(ObjectViewerPage).attrs({
  className: ObjectViewerPage.name,
})``;
