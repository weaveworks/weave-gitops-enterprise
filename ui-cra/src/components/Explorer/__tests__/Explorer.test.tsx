import {
  RenderResult,
  act,
  queryByLabelText,
  render,
} from '@testing-library/react';
import QueryServiceProvider from '../../../contexts/QueryService';
import {
  MockQueryService,
  defaultContexts,
  withContext,
} from '../../../utils/test-utils';
import Explorer from '../Explorer';

describe('Explorer', () => {
  let wrap: (el: JSX.Element) => JSX.Element;
  let api: MockQueryService;

  beforeEach(() => {
    api = new MockQueryService();
    wrap = withContext([...defaultContexts(), [QueryServiceProvider, { api }]]);
  });

  it('renders rows', async () => {
    const objects = [
      {
        kind: 'Kustomization',
        name: 'flux-system',
        namespace: 'flux-system',
        status: 'Ready',
      },
      {
        kind: 'HelmRelease',
        name: 'flux-system',
        namespace: 'flux-system',
        status: 'Ready',
      },
    ];

    api.DoQueryReturns = {
      objects,
    };

    let result: RenderResult;
    await act(async () => {
      const c = wrap(<Explorer />);
      result = await render(c);
    });

    // @ts-ignore
    expect(result.container).toHaveTextContent(objects[0].name);
  });
  it('renders filter elements', async () => {
    api.DoQueryReturns = {
      objects: [],
    };

    const facets = [
      {
        field: 'Kind',
        values: ['Kustomization', 'HelmRelease', 'GitRepository'],
      },
    ];

    api.ListFacetsReturns = {
      facets,
    };

    let result: RenderResult;

    await act(async () => {
      const c = wrap(<Explorer />);
      result = await render(c);
    });

    // @ts-ignore
    expect(result.container).toHaveTextContent(facets[0].field);
    // @ts-ignore
    expect(result.container).toHaveTextContent(facets[0].values[0]);
  });
  it('renders filter state', async () => {
    api.DoQueryReturns = {
      objects: [],
    };

    const facets = [
      {
        field: 'Kind',
        values: ['Kustomization', 'HelmRelease', 'GitRepository'],
      },
    ];

    api.ListFacetsReturns = {
      facets,
    };

    const manager = {
      read: jest.fn(() => ({
        terms: '',
        filters: ['Kind:Kustomization'],
        limit: 0,
        offset: 0,
        orderBy: '',
        orderAscending: false,
      })),

      write: jest.fn(),
    };

    let result: RenderResult;

    await act(async () => {
      const c = wrap(<Explorer manager={manager} />);
      result = await render(c);
    });

    //  @ts-ignore
    const container = result.container;

    const input = queryByLabelText(
      container,
      'Kustomization',
    ) as HTMLInputElement;

    expect(input?.checked).toBeTruthy();
  });

  describe('snapshots', () => {
    it('renders', async () => {
      let result: RenderResult;

      api.DoQueryReturns = {
        objects: [
          {
            kind: 'Kustomization',
            name: 'flux-system',
            namespace: 'flux-system',
            status: 'Ready',
          },
        ],
      };
      api.ListFacetsReturns = {
        facets: [
          {
            field: 'Kind',
            values: ['Kustomization', 'HelmRelease', 'GitRepository'],
          },
        ],
      };

      await act(async () => {
        const c = wrap(<Explorer />);
        result = await render(c);
      });

      //   @ts-ignore
      expect(result.container).toMatchSnapshot();
    });
  });
});
