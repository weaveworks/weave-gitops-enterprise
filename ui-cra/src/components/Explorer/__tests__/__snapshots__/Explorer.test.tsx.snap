// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explorer snapshots renders 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  height: 100%;
  width: 100%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  height: 100%;
  width: 100%;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c17 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  text-transform: none;
  font-style: normal;
}

.c19 {
  font-family: 'proxima-nova',Helvetica,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  text-transform: none;
  font-style: normal;
  color: #00b3ec;
}

.c5 svg {
  fill: #737373;
}

.c5 svg path.path-fill,
.c5 svg line.path-fill,
.c5 svg polygon.path-fill,
.c5 svg rect.path-fill,
.c5 svg circle.path-fill,
.c5 svg polyline.path-fill {
  fill: #737373 !important;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c5 svg path.stroke-fill,
.c5 svg line.stroke-fill,
.c5 svg polygon.stroke-fill,
.c5 svg rect.stroke-fill,
.c5 svg circle.stroke-fill,
.c5 svg polyline.stroke-fill {
  stroke: #737373 !important;
  -webkit-transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c5.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c5.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c5 .c16 {
  margin-left: 4px;
  color: #737373;
}

.c15 svg {
  height: 16px;
  width: 16px;
}

.c15 svg path.path-fill,
.c15 svg line.path-fill,
.c15 svg polygon.path-fill,
.c15 svg rect.path-fill,
.c15 svg circle.path-fill,
.c15 svg polyline.path-fill {
  fill: !important;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c15 svg path.stroke-fill,
.c15 svg line.stroke-fill,
.c15 svg polygon.stroke-fill,
.c15 svg rect.stroke-fill,
.c15 svg circle.stroke-fill,
.c15 svg polyline.stroke-fill {
  stroke: !important;
  -webkit-transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c15 svg rect.rect-height {
  height: 16px;
  width: 16px;
}

.c15.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c15.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c15 .c16 {
  margin-left: 4px;
}

.c15 img {
  width: 16px;
}

.c28 .c16 {
  margin-left: 4px;
  color: #d8d8d8;
}

.c25 svg path.path-fill,
.c25 svg line.path-fill,
.c25 svg polygon.path-fill,
.c25 svg rect.path-fill,
.c25 svg circle.path-fill,
.c25 svg polyline.path-fill {
  fill: !important;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c25 svg path.stroke-fill,
.c25 svg line.stroke-fill,
.c25 svg polygon.stroke-fill,
.c25 svg rect.stroke-fill,
.c25 svg circle.stroke-fill,
.c25 svg polyline.stroke-fill {
  stroke: !important;
  -webkit-transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c25.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c25.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c25 .c16 {
  margin-left: 4px;
}

.c20 svg {
  fill: #BC3B1D;
}

.c20 svg path.path-fill,
.c20 svg line.path-fill,
.c20 svg polygon.path-fill,
.c20 svg rect.path-fill,
.c20 svg circle.path-fill,
.c20 svg polyline.path-fill {
  fill: #BC3B1D !important;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c20 svg path.stroke-fill,
.c20 svg line.stroke-fill,
.c20 svg polygon.stroke-fill,
.c20 svg rect.stroke-fill,
.c20 svg circle.stroke-fill,
.c20 svg polyline.stroke-fill {
  stroke: #BC3B1D !important;
  -webkit-transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: stroke 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
}

.c20.downward {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c20.upward {
  -webkit-transform: initial;
  -ms-transform: initial;
  transform: initial;
}

.c20 .c16 {
  margin-left: 4px;
  color: #BC3B1D;
}

.c14 {
  padding: 4px;
}

.c18 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  height: 40px;
  padding: 4px 0px;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  overflow-x: auto;
}

.c3 .MuiChip-root {
  color: #1a1a1a;
  margin-right: 4px;
  background-color: #F6F7F9;
  -webkit-transition: none;
  transition: none;
}

.c13.MuiButton-root {
  margin: 0;
  text-transform: none;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.c13.MuiButton-text {
  min-width: 0px;
}

.c13.MuiButton-text .selected {
  color: #1a1a1a;
}

.c13.arrow {
  min-width: 0px;
}

.c10 {
  max-width: 100%;
}

.c8 {
  width: 100%;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  overflow-x: hidden;
}

.c8 h2 {
  padding: 8px;
  font-size: 12px;
  font-weight: 600;
  color: #737373;
  margin: 0px;
  white-space: nowrap;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c8 .MuiTableRow-root {
  -webkit-transition: background 0.5s ease-in-out;
  transition: background 0.5s ease-in-out;
}

.c8 .MuiTableRow-root:not(.MuiTableRow-head):hover {
  background: #f5f5f5;
  -webkit-transition: background 0.5s ease-in-out;
  transition: background 0.5s ease-in-out;
}

.c8 .MuiTableCell-root {
  border-color: #d8d8d8;
}

.c8 table {
  margin-top: 12px;
}

.c8 th {
  padding: 0;
  background: #F6F7F9;
}

.c8 th .MuiCheckbox-root {
  padding: 4px 9px;
}

.c8 th:first-child {
  border-top-left-radius: 4px;
}

.c8 th:last-child {
  border-top-right-radius: 4px;
}

.c8 td {
  padding-left: 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c9 {
  width: 100%;
}

.c9 td:nth-child(6),
.c9 td:nth-child(7) {
  white-space: pre-wrap;
  overflow-wrap: break-word;
  word-wrap: break-word;
}

.c9 > div:first-child {
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c22 {
  position: relative;
  height: 100%;
  -webkit-transition-property: width,left;
  transition-property: width,left;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-timing-function: linear;
  transition-timing-function: linear;
  overflow: hidden;
  width: 0px;
  -webkit-flex: 2;
  -ms-flex: 2;
  flex: 2;
  left: 24px;
}

.c22.open {
  left: 0;
  width: 360px;
}

.c23 {
  height: 100%;
  width: 100%;
  border-left: 2px solid #d8d8d8;
  padding-left: 32px;
}

.c21 {
  overflow: hidden;
  white-space: nowrap;
}

.c26 ul {
  padding: 0px 8px;
}

.c26 h2 {
  margin-top: 0;
}

.c26 h3 {
  margin-bottom: 0;
}

.c24 {
  position: relative;
}

.c0 {
  width: 100%;
}

<div>
  <div
    class="c0 Explorer"
  >
    <div
      class="c1"
    >
      <div
        class="QueryStateChips"
      >
        <div
          class="c2 c3 $9121bed0097d59fd$var$ChipGroup"
        />
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <div
            class="c4 c5"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
              />
            </svg>
          </div>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="c6"
    >
      <div
        class="c7 c8 c9 ExplorerTable"
      >
        <div
          class="c1 c10"
        />
        <div
          class="c11"
        >
          <div
            class="MuiTableContainer-root"
          >
            <table
              aria-label="simple table"
              class="MuiTable-root"
            >
              <thead
                class="MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    <div
                      class="c12"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text c13 MuiButton-colorInherit"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <h2
                            class="selected"
                          >
                            Name
                          </h2>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <div
                        class="c14"
                      />
                      <div
                        class="c4 c15 downward"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                          />
                        </svg>
                      </div>
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    <div
                      class="c12"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text c13 MuiButton-colorInherit"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <h2
                            class=""
                          >
                            Kind
                          </h2>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <div
                        class="c14"
                      />
                      <div
                        style="width: 16px;"
                      />
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    <div
                      class="c12"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text c13 MuiButton-colorInherit"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <h2
                            class=""
                          >
                            Namespace
                          </h2>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <div
                        class="c14"
                      />
                      <div
                        style="width: 16px;"
                      />
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    <div
                      class="c12"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text c13 MuiButton-colorInherit"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <h2
                            class=""
                          >
                            Cluster
                          </h2>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <div
                        class="c14"
                      />
                      <div
                        style="width: 16px;"
                      />
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    <div
                      class="c12"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text c13 MuiButton-colorInherit"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <h2
                            class=""
                          >
                            Status
                          </h2>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <div
                        class="c14"
                      />
                      <div
                        style="width: 16px;"
                      />
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    <div
                      class="c12"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text c13 MuiButton-colorInherit"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          <h2
                            class=""
                          >
                            Message
                          </h2>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                      <div
                        class="c14"
                      />
                      <div
                        style="width: 16px;"
                      />
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                  >
                    <span
                      class="c16 c17"
                    >
                      <a
                        class="c18 $8405a4f7caf64b1d$var$Link"
                        href="/kustomization?name=flux-system&namespace=flux-system"
                      >
                        <span
                          class="c16 c19"
                          color="primary"
                        >
                          flux-system
                        </span>
                      </a>
                    </span>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                  >
                    <span
                      class="c16 c17"
                    >
                      Kustomization
                    </span>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                  >
                    <span
                      class="c16 c17"
                    >
                      flux-system
                    </span>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                  >
                    <span
                      class="c16 c17"
                    >
                      -
                    </span>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                  >
                    <span
                      class="c16 c17"
                    >
                      <div
                        class="c4"
                      >
                        <div
                          class="MuiBox-root MuiBox-root-19"
                        >
                          <div
                            class="c4 c20"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
                              />
                            </svg>
                          </div>
                        </div>
                        Ready
                      </div>
                    </span>
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body"
                  >
                    <span
                      class="c16 c17"
                    >
                      -
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        class="c21 FilterDrawer"
      >
        <div
          class="c22"
        >
          <div
            class="c23"
          >
            <div
              class="MuiBox-root MuiBox-root-17"
            >
              <div>
                <h2>
                  Filters
                </h2>
              </div>
              <div>
                <div
                  class="c6 c24 QueryInput"
                >
                  <div
                    class="c4"
                  >
                    <div
                      class="c4 c25"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiFormControl-root"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root $66a683f19021b9f2$var$Input"
                      >
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiInput-input"
                            placeholder="Search"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="c26 Filters"
                >
                  <div>
                    <h3>
                      Kind
                    </h3>
                    <ul
                      style="list-style: none;"
                    >
                      <li>
                        <label
                          class="MuiFormControlLabel-root"
                        >
                          <span
                            aria-disabled="false"
                            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <input
                                class="PrivateSwitchBase-input-23"
                                data-indeterminate="false"
                                type="checkbox"
                                value=""
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                          >
                            Kustomization
                          </span>
                        </label>
                      </li>
                      <li>
                        <label
                          class="MuiFormControlLabel-root"
                        >
                          <span
                            aria-disabled="false"
                            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <input
                                class="PrivateSwitchBase-input-23"
                                data-indeterminate="false"
                                type="checkbox"
                                value=""
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                          >
                            HelmRelease
                          </span>
                        </label>
                      </li>
                      <li>
                        <label
                          class="MuiFormControlLabel-root"
                        >
                          <span
                            aria-disabled="false"
                            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-20 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <input
                                class="PrivateSwitchBase-input-23"
                                data-indeterminate="false"
                                type="checkbox"
                                value=""
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </span>
                          <span
                            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                          >
                            GitRepository
                          </span>
                        </label>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c27 PaginationControls"
    >
      <div
        class="MuiBox-root MuiBox-root-18"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <div
              class="c4 c25"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </div>
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <div
              class="c4 c25"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </div>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
